pkgs/applications/editors/emacs-25/macport-25.1.nix

```diff
@@ -0,0 +1,90 @@
+{ stdenv, fetchurl, ncurses, pkgconfig, texinfo, libxml2, gnutls, gettext, autoconf, automake
+, AppKit, Carbon, Cocoa, IOKit, OSAKit, Quartz, QuartzCore, WebKit
+, ImageCaptureCore, GSS, ImageIO # These may be optional
+}:
+
+stdenv.mkDerivation rec {
+  emacsName = "emacs-25.1-rc2";
+  name = "${emacsName}-mac-5.91";
+
+  builder = ./builder.sh;
+
+  src = fetchurl {
+    url = "ftp://alpha.gnu.org/gnu/emacs/pretest/${emacsName}.tar.xz";
+    sha256 = "1hffvyvl50mrivdv6lp92sbxi3l2zhblj8npmpbzk47zpl1mzm2v";
+  };
+
+  macportSrc = fetchurl {
+    url = "ftp://ftp.math.s.chiba-u.ac.jp/emacs/${name}.tar.gz";
+    sha256 = "1326f2d7d9b2e9a0cd403747be10d072f5f7f6acfa7ab3a1a0b9497c3b792c21";
+  };
+
+  enableParallelBuilding = true;
+
+  buildInputs = [ ncurses libxml2 gnutls pkgconfig texinfo gettext autoconf automake];
+
+  propagatedBuildInputs = [
+    AppKit Carbon Cocoa IOKit OSAKit Quartz QuartzCore WebKit
+    ImageCaptureCore GSS ImageIO   # may be optional
+  ];
+
+  postUnpack = ''
+    mv $sourceRoot $name
+    tar xzf $macportSrc
+    mv $name $sourceRoot
+  '';
+
+  postPatch = ''
+    patch -p1 < patch-mac
+    sed -i 's|/usr/share/locale|${gettext}/share/locale|g' lisp/international/mule-cmds.el
+  '';
+
+  configureFlags = [
+    "LDFLAGS=-L${ncurses.out}/lib"
+    "--with-xml2=yes"
+    "--with-gnutls=yes"
+    "--with-mac"
+    "--enable-mac-app=$$out/Applications"
```

Comment:
Since this package port is a parallel port of macport-24.5.nix, i think the original author's intention is to reference a shell variable inside configure/make.